### PR TITLE
A command to bump package version

### PR DESCRIPTION
**Pull request by alexander-fenster**
I feel I need to automate bumping the version as having samples pointing to the wrong version of the main package is a very common problem (I fixed it in all our packages but don't want it to appear again). Having a good automated way of taking care of that might help.

Note: I don't really know the structure or the coding style of `nodejs-repo-tools` so please feel free to tell me if anything needs to be done in a different way :)

### CODE REVIEW

**Comment by alexander-fenster** [comment]
Sounds reasonable, I will do it.

**Comment by alexander-fenster** [comment]
@jmdobry I feel I might need your help though. I suppose to make the command language agnostic it should be somehow built into the build pack (e.g. `src/build_packs/nodejs.js`) but there is no good place to add a custom function to a build pack. Also, I see there is one command specific to Node.js:
```
  unify                 (Node.js only) Recursively add sub-directory dependencies to the top-level package.json file.
```

I see two options here:
1. Use build packs:
- define the `bump` command in `src/build_packs/build_pack.js`,
- move the Node.js-specific version increment code to a separate file (e.g. `src/build_packs/impl/nodejs_bump.js`,
- require it from `src/build_packs/nodejs.js`,
- override the `bump` command in `src/build_packs/nodejs.js` to calling the bump code;
2. Leave as is, adding comments that the command is Node.js-specific (just like `unify`).

What do you think? Am I missing any other good way of implementing it?

**Comment by jmdobry** [comment]
Add the Node.js-specific bits of the generic `bump` command to the [NodejsBuildPack](https://github.com/GoogleCloudPlatform/nodejs-repo-tools/blob/master/src/build_packs/nodejs.js#L186) class, under a `bump` instance method. Then use the instance method in the `bump` CLI code.

**Comment by alexander-fenster** [comment]
OK - I thought it would be too huge to include right in that class, that's why I wanted to put it into a separate file, but I sure can do it this way.

**Comment by jmdobry** [comment]
We can refactor those classes as they grow.

**Comment by alexander-fenster** [comment]
@jmdobry Updated, PTAL.

File: src/build_packs/nodejs.js [hunk 1]
```diff
@@ -207,4 +211,114 @@ module.exports = class NodejsBuildPack extends BuildPack {
...
+        `${'samples/package.json'.yellow} will depend on ${dependency.yellow}.`
+      );
+    }
+
+    if (opts.dryRun) {
+      utils.logger.log('bump', 'Dry run complete.'.cyan);
+      return;
+    }
+
+    let newPackageJson = extend(true, {}, this.config.pkgjson);
```
Comment (jmdobry) [inline src/build_packs/nodejs.js]: I believe there's a method in lodash you can use to do this, and you can eliminate the dependency on `extend`.

Comment (alexander-fenster) [reply src/build_packs/nodejs.js]: I probably can just edit `this.config.pkgjson` without copying it to the new object - originally I was not sure if it's going to break anything, but I now think it should be perfectly OK. What do you think?

Comment (jmdobry) [reply src/build_packs/nodejs.js]: sgtm

**Comment by jmdobry** [comment]
Probably would be good to have some system tests for this feature, as it modifies existing files.

**Comment by alexander-fenster** [comment]
✔ cli › nodejs › bump › should do a dry run bump patch with samples (1.8s)
  ✔ cli › nodejs › bump › should do a dry run bump minor with samples (509ms)
  ✔ cli › nodejs › bump › should do a dry run bump major with samples (541ms)
  ✔ cli › nodejs › bump › should bump patch with samples (531ms)
  ✔ cli › nodejs › bump › should bump minor with samples (545ms)
  ✔ cli › nodejs › bump › should bump major with samples (525ms)
  ✔ cli › nodejs › bump › should work without samples (651ms)

**Comment by alexander-fenster** [comment]
@jmdobry can you please take a look?

**Comment by alexander-fenster** [comment]
@jmdobry Ping on this :) Thanks!

**Comment by alexander-fenster** [comment]
Thanks @jmdobry !

**Pull request merged**